src/casts/specify-representation.rkt

```diff
@@ -122,76 +122,73 @@ but a static single assignment is implicitly maintained.
      (begin$
        (assign$ val e1)
        (assign$ type (sr-expr e2))
-       (assign$ tag (sr-get-tag type TYPE-TAG-MASK))
-       (Switch
-           type
-         (let* ([shifted-imm (op$ %<< val DYN-IMDT-SHIFT)]
-                [tag-shifted-imm
-                 (lambda ([tag : D0-Expr])
-                   : D0-Expr
-                   (op$ binary-or shifted-imm tag))])
-           `([(,data:TYPE-INT-RT-VALUE)  . ,(tag-shifted-imm DYN-INT-TAG)]
-             [(,data:TYPE-BOOL-RT-VALUE) . ,(tag-shifted-imm DYN-BOOL-TAG)]
-             [(,data:TYPE-UNIT-RT-VALUE) . ,(tag-shifted-imm DYN-UNIT-TAG)]
-             [(,data:TYPE-CHAR-RT-VALUE) . ,(tag-shifted-imm DYN-CHAR-TAG)]))
-         ;; Notice that float types fall into this case also
-         (sr-alloc "dynamic_boxed" DYN-BOXED-TAG `(("" . ,val) ("" . ,type)))))]))
+       (let* ([shifted-imm (op$ %<< val DYN-IMDT-SHIFT)]
+              [tag-shifted-imm
+               (lambda ([tag : D0-Expr])
+                 : D0-Expr
+                 (op$ binary-or shifted-imm tag))])
+         (cond$
+          [(atomic-type? type)
+           (case$ type
+             [(data:TYPE-INT-RT-VALUE) (tag-shifted-imm DYN-INT-TAG)]
+             [(data:TYPE-BOOL-RT-VALUE) (tag-shifted-imm DYN-BOOL-TAG)]
+             [(data:TYPE-UNIT-RT-VALUE) (tag-shifted-imm DYN-UNIT-TAG)]
+             [(data:TYPE-CHAR-RT-VALUE) (tag-shifted-imm DYN-CHAR-TAG)]
+             [(data:TYPE-FLOAT-RT-VALUE)
+              (sr-alloc
+               "dynamic_boxed" DYN-BOXED-TAG `(("" . ,val) ("" . ,type)))]
+             [else
+              (op$ Print (Quote "bad base type"))
+              (op$ Exit (Quote -1)) UNDEF-IMDT])]
```

Comment:
Do we have a standard grift screwed up exit code? I only ask because there are two different codes used in this PR.

Reply:
I do not believe so, I agree we should standardize it.

Reply:
Done in a new commit.